R/bs_sass.R

```diff
@@ -1,32 +1,40 @@
-#' or mixins) to generate additional CSS rules (as a string that can be included
-#' as a `<style>` tag via `tags$style(css)`).
+#' Use `bootstrap()` to compile Bootstrap Sass into CSS and return it, along
+#' with other HTML dependencies, as a list of [htmltools::htmlDependency()]s (if
+#' you just want the CSS as a string, and a [bs_theme_new()] has been set, you
+#' can call `sass::sass()` on the return value of [bs_theme_get()]). Use
+#' `bootstrap_sass()` if you can assume Bootstrap already exists on the page,
+#' but you want to leverage Bootstrap utilities (e.g., variables, functions, or
+#' mixins) to generate additional CSS rules (as a string that can be included as
+#' a `<style>` tag via `tags$style(css)`).
 #'
-#' @param theme one of the following:
-#'   1. The result of [bs_theme_get()] (i.e., the current global theme).
-#'   2. `NULL`, which means use the latest version of Bootstrap with no custom theming.
-#'   3. A string containing a bootswatch theme and/or a Bootstrap major version. To specify
-#'   both, use the syntax `"theme@version"`, (e.g., `"cosmo@4"` for Bootstrap 4 cosmo
-#'   theme with BS3 compatibility). If no version is specified, the latest available
-#'   version is used (for more info, see `version` in [bs_theme_new()]).
-#'   __Note__: this approach ignores global themes (i.e., [bs_theme_new()])
-#'   4. A [sass::sass_layer()] which contains a bootstraplib theme. Useful for adding
-#'   custom layers to the current theme without affecting the global state (e.g.,
-#'   `sass::sass_layer_merge(bs_theme_get(), my_layer())`).
+#' @param theme one of the following: 1. The result of [bs_theme_get()] (i.e.,
+#'   the current global theme). 2. `NULL`, which means use the latest version of
+#'   Bootstrap with no custom theming. 3. A string containing a bootswatch theme
+#'   and/or a Bootstrap major version. To specify both, use the syntax
+#'   `"theme@version"`, (e.g., `"cosmo@4"` for Bootstrap 4 cosmo theme with BS3
+#'   compatibility). If no version is specified, the latest available version is
+#'   used (for more info, see `version` in [bs_theme_new()]). __Note__: this
+#'   approach ignores global themes (i.e., [bs_theme_new()]) 4. A
+#'   [sass::sass_layer()] which contains a bootstraplib theme. Useful for adding
+#'   custom layers to the current theme without affecting the global state
+#'   (e.g., `sass::sass_layer_merge(bs_theme_get(), my_layer())`).
 #' @param sass_options see [sass::sass_options()].
 #' @param jquery See [jquerylib::jquery_core()].
+#' @param use_precompiled_css Before compiling the theme object, first look for
+#'   a precompiled CSS file for the given `theme` and `sass_options`. The
+#'   bootstraplib comes with precompiled CSS files for stock Bootstrap builds
+#'   for when `bs_theme_new()` or `bs_theme_create()` are called with `version`
+#'   of `"3"`, `"4+3"`, and `"4"`, and no `bootswatch` theme. If this option is
```

Comment:
I'd like to avoid listing supported versions in multiple place (to make it easier to add new versions)

Reply:
I don't think the set of precompiled versions is listed elsewhere, is it?

Reply:
I just mean generally...when we go to add Bootstrap 5 I'd like that to be as seamless as possible. How about we say something like

```r
#' @param use_precompiled_css Before compiling the theme object, first look for
#'   a precompiled CSS file for the given `version`. At the moment, we only provide
#'   precompiled CSS for "stock" builds of Bootstrap (i.e., no theming additions or
#'   non-default `sass_options`).
```